src/cmd/delegation_backend/main_bpu.go

```diff
@@ -25,29 +26,48 @@ func main() {
 	log := logging.Logger("delegation backend")
 	log.Infof("delegation backend has the following logging subsystems active: %v", logging.GetSubsystems())
 
+	// Context and app initialization
 	ctx := context.Background()
-
 	appCfg := LoadEnv(log)
+	app := new(App)
+	app.Log = log
 
-	awsCfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(appCfg.Aws.Region))
-	if err != nil {
-		log.Fatalf("Error loading AWS configuration: %v", err)
+	// Storage backend setup
+	if appCfg.Aws != nil {
+		log.Infof("storage backend: AWS S3")
+		awsCfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(appCfg.Aws.Region))
+		if err != nil {
+			log.Fatalf("Error loading AWS configuration: %v", err)
+		}
+		client := s3.NewFromConfig(awsCfg)
+		awsctx := AwsContext{Client: client, BucketName: aws.String(GetAWSBucketName(appCfg)), Prefix: appCfg.NetworkName, Context: ctx, Log: log}
+		app.Save = func(objs ObjectsToSave) {
+			awsctx.S3Save(objs)
+		}
+	} else if appCfg.LocalFileSystem != nil {
+		log.Infof("storage backend: Local File System")
+		// app.Save = func(objs ObjectsToSave) {
+		// 	LocalFileSystemSave(objs, appCfg.LocalFileSystem.Path, log)
```

Comment:
This function doesn't exist yet, it's just to show how it might look like. Both LocalFileSystemSave and DatabaseSave are expected to arrive in next prs for this story.

Reply:
In that case maybe it makes sense to put there a comment explaining that. Without it it just looks like a leftover from development process.